lib/sprockets/manifest.rb

```diff
@@ -285,6 +275,20 @@ def save
     end
 
     private
+
+      def compress_asset(asset, target)
+        return if environment.skip_gzip?
+        gzip = Utils::Gzip.new(asset)
+        return if gzip.cannot_compress?(environment.mime_types)
+
+        if File.exist?("#{target}.gz")
+          logger.debug "Skipping #{target}.gz, already exists"
```

Comment:
There's no mutex around the logging in sprockets so logging from within an thread can result in garbled output. That's why originally it was logged outside of the pool.